libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/lifecycle/UIComponent.kt

```diff
@@ -35,3 +35,32 @@ open class UIComponent : MapboxNavigationObserver {
         coroutineScope.launch { collect(action) }
     }
 }
+
+/**
+ * Return UIComponent that gets re-created using [factory] when [value] changes.
+ *
+ * eg.
+ * ```
+ *   reloadOnChange(myFlowValue) { value ->
+ *      MyComponent(value)
+ *   }
+ * ```
+ */
+fun <T> reloadOnChange(value: Flow<T>, factory: (value: T) -> UIComponent): UIComponent =
+    object : UIComponent() {
+        private var childComponent: UIComponent? = null
+
+        override fun onAttached(mapboxNavigation: MapboxNavigation) {
+            super.onAttached(mapboxNavigation)
+            value.observe {
+                childComponent?.onDetached(mapboxNavigation)
+                childComponent = factory(it)
+                childComponent?.onAttached(mapboxNavigation)
+            }
+        }
+
+        override fun onDetached(mapboxNavigation: MapboxNavigation) {
+            super.onDetached(mapboxNavigation)
+            childComponent?.onDetached(mapboxNavigation)
+        }
+    }
```

Comment:
Awesome! This is promising. Will you add some tests?

You should set childComponent to null in `onDetached`
``` kotlin
        override fun onDetached(mapboxNavigation: MapboxNavigation) {
            super.onDetached(mapboxNavigation)
            childComponent?.onDetached(mapboxNavigation)
            childComponent = null
        }
```

Another way to make it so you don't need to null out the child, is to make it just fall out of scope.

```suggestion
fun <T> reloadOnChange(value: Flow<T>, factory: (value: T) -> UIComponent) =
    object : UIComponent() {
        override fun onAttached(mapboxNavigation: MapboxNavigation) {
            super.onAttached(mapboxNavigation)
            var childComponent: UIComponent? = null
            coroutineScope.launch {
                value.collect { value ->
                    childComponent?.onDetached(mapboxNavigation)
                    childComponent = factory(value).also {
                        onAttached(mapboxNavigation)
                    }
                }
            }.invokeOnCompletion {
                childComponent?.onDetached(mapboxNavigation)
            }
        }
    }
```

Reply:
Done.